In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
cd '/content/drive/MyDrive/DSGA_1011_NLP_Project/Data/cnndm'

/content/drive/.shortcut-targets-by-id/1zJEzU7AONU-4f_78lpneSWk_NcBn0nda/DSGA_1011_NLP_Project/Data/cnndm


In [ ]:
!pip install torch==1.6.0

     |████████████████████████████████| 748.8MB 22kB/s 
ERROR: torchvision 0.8.1+cu101 has requirement torch==1.7.0, but you'll have torch 1.6.0 which is incompatible.
  Found existing installation: torch 1.7.0+cu101
    Uninstalling torch-1.7.0+cu101:
      Successfully uninstalled torch-1.7.0+cu101


In [ ]:
!git clone https://github.com/OpenNMT/OpenNMT-py.git

Cloning into 'OpenNMT-py'...
remote: Enumerating objects: 41, done.
remote: Counting objects: 100% (41/41), done.
remote: Compressing objects: 100% (41/41), done.
remote: Total 16803 (delta 0), reused 5 (delta 0), pack-reused 16762
Receiving objects: 100% (16803/16803), 223.31 MiB | 19.14 MiB/s, done.
Resolving deltas: 100% (12094/12094), done.
Checking out files: 100% (225/225), done.


In [ ]:
cd OpenNMT-py

/content/drive/.shortcut-targets-by-id/1zJEzU7AONU-4f_78lpneSWk_NcBn0nda/DSGA_1011_NLP_Project/Data/cnndm/OpenNMT-py


In [ ]:
!git checkout a0095fa2

error: Your local changes to the following files would be overwritten by checkout:
	examples/scripts/prepare_wmt_data.sh
	onmt/bin/average_models.py
	onmt/bin/release_model.py
	onmt/bin/server.py
	onmt/bin/translate.py
	onmt/tests/pull_request_chk.sh
	onmt/tests/rebuild_test_models.sh
	tools/average_models.py
	tools/embeddings_to_torch.py
Please commit your changes or stash them before you switch branches.
Aborting


In [ ]:
cd ..

/content/drive/.shortcut-targets-by-id/1zJEzU7AONU-4f_78lpneSWk_NcBn0nda/DSGA_1011_NLP_Project/Data/cnndm


In [ ]:
!pip install -r OpenNMT-py/requirements.txt

  Cloning https://github.com/pytorch/text to /tmp/pip-req-build-yrbbr5c0
  Running command git clone -q https://github.com/pytorch/text /tmp/pip-req-build-yrbbr5c0
  Running command git submodule update --init --recursive -q
  Created wheel for torchtext: filename=torchtext-0.9.0a0+d6d7f20-cp36-cp36m-linux_x86_64.whl size=6916062 sha256=52be98a68e3dfe5dc736a823bf5a7bbf5ab3ed5341b219a4aa55af85cac17f6e
  Stored in directory: /tmp/pip-ephem-wheel-cache-y1yz19u4/wheels/73/14/71/ed033fd999ae4933e17df3e91be2014e61c2f312a88a164ff5
Successfully built torchtext
  Found existing installation: torchtext 0.3.1
    Uninstalling torchtext-0.3.1:
      Successfully uninstalled torchtext-0.3.1


In [ ]:
import pandas as pd
import numpy as np

def open_file(filename):
    with open(filename, encoding='utf-8') as file:
      text = file.read()
    return text.strip().split('\n')

def get_data(name):
  src_path = name+'.txt.src'
  data = open_file(src_path)
  df = pd.DataFrame(data,columns=['source'])
  target_path = name+'.txt.tgt.tagged'
  target = open_file(target_path)
  df['target'] = pd.DataFrame(target)
  print(f"{name} Data Size: {df.shape}")
  df['len_source']= df['source'].apply(lambda x:len(x))
  df['len_target']= df['target'].apply(lambda x:len(x))
  print(df.head())
  return df

train = get_data('train')
test = get_data('test')
val = get_data('val')

train Data Size: (287227, 2)
                                              source  ... len_target
0  editor 's note : in our behind the scenes seri...  ...        332
1  london , england -lrb- reuters -rrb- -- harry ...  ...        246
2  minneapolis , minnesota -lrb- cnn -rrb- -- dri...  ...        274
3  baghdad , iraq -lrb- cnn -rrb- -- dressed in a...  ...        290
4  washington -lrb- cnn -rrb- -- doctors removed ...  ...        218

[5 rows x 4 columns]
test Data Size: (11490, 2)
                                              source  ... len_target
0  marseille , france -lrb- cnn -rrb- the french ...  ...        349
1  -lrb- cnn -rrb- the palestinian authority offi...  ...        252
2  -lrb- cnn -rrb- governments around the world a...  ...        378
3  -lrb- cnn -rrb- on may 28 , 2014 , some 7,000 ...  ...        237
4  -lrb- cnn -rrb- seventy years ago , anne frank...  ...        230

[5 rows x 4 columns]
val Data Size: (13368, 2)
                                              

In [ ]:
src_path = 'test.txt.src'
data = open_file(src_path)
len(data)

11490

In [ ]:
data[11489]

"despite the hype surrounding its first watch , the iphone is still the engine behind apple 's phenomenal success , its latest figures have revealed . the results far surpassed most analysts ' expectations for the first three months of the year , when sales traditionally fall from their holiday-season peak . apple sold more than 61 million iphones in the quarter , accounting for more than two-thirds of its $ 58 billion in revenue for the quarter and the lion 's share of its $ 13.6 billion in profit - and up 40 % from a year ago . sales of iphones in china were also revealed to have outstripped those in the us . apple sold more than 61 million iphones in the quarter , accounting for more than two-thirds of its $ 58 billion in revenue for the quarter and the lion 's share of its $ 13.6 billion in profit . $ 58 billion in revenue , $ 13.6 billion in profit . $ 200 billion in cash , up from around $ 150 billion a year ago . more than 61 million iphones sole . ipad revenue fell 29 % to $ 5.

In [ ]:
data=data[5680:]
len(data)

5810

In [ ]:
with open("remaining_test.txt.src",'w') as f: 
  f.write('\n'.join(data))

The CNN / Daily Mail dataset as processed by Nallapati et al. (2016) has been used for evaluating summarization. The dataset contains online news articles (781 tokens on average) paired with multi-sentence summaries (3.75 sentences or 56 tokens on average). The processed version contains 287,226 training pairs, 13,368 validation pairs and 11,490 test pairs.

In [ ]:
train.loc[0,'source']

"editor 's note : in our behind the scenes series , cnn correspondents share their experiences in covering news and analyze the stories behind the events . here , soledad o'brien takes users inside a jail where many of the inmates are mentally ill . an inmate housed on the `` forgotten floor , '' where many mentally ill inmates are housed in miami before trial . miami , florida -lrb- cnn -rrb- -- the ninth floor of the miami-dade pretrial detention facility is dubbed the `` forgotten floor . '' here , inmates with the most severe mental illnesses are incarcerated until they 're ready to appear in court . most often , they face drug charges or charges of assaulting an officer -- charges that judge steven leifman says are usually `` avoidable felonies . '' he says the arrests often result from confrontations with police . mentally ill people often wo n't do what they 're told when police arrive on the scene -- confrontation seems to exacerbate their illness and they become more paranoid 

In [ ]:
train.loc[0,'target']

"<t> mentally ill inmates in miami are housed on the `` forgotten floor '' </t> <t> judge steven leifman says most are there as a result of `` avoidable felonies '' </t> <t> while cnn tours facility , patient shouts : `` i am the son of the president '' </t> <t> leifman says the system is unjust and he 's fighting for change . </t>"

In [ ]:
train.loc[1,'source']

"london , england -lrb- reuters -rrb- -- harry potter star daniel radcliffe gains access to a reported # 20 million -lrb- $ 41.1 million -rrb- fortune as he turns 18 on monday , but he insists the money wo n't cast a spell on him . daniel radcliffe as harry potter in `` harry potter and the order of the phoenix '' to the disappointment of gossip columnists around the world , the young actor says he has no plans to fritter his cash away on fast cars , drink and celebrity parties . `` i do n't plan to be one of those people who , as soon as they turn 18 , suddenly buy themselves a massive sports car collection or something similar , '' he told an australian interviewer earlier this month . `` i do n't think i 'll be particularly extravagant . `` the things i like buying are things that cost about 10 pounds -- books and cds and dvds . '' at 18 , radcliffe will be able to gamble in a casino , buy a drink in a pub or see the horror film `` hostel : part ii , '' currently six places below hi

In [ ]:
train.loc[1,'target']

"<t> harry potter star daniel radcliffe gets # 20m fortune as he turns 18 monday . </t> <t> young actor says he has no plans to fritter his cash away . </t> <t> radcliffe 's earnings from first five potter films have been held in trust fund . </t>"

In [ ]:
import torch
torch.__version__

'1.6.0'

In [ ]:
!git clone https://github.com/OpenNMT/OpenNMT-py.git

fatal: destination path 'OpenNMT-py' already exists and is not an empty directory.


In [ ]:
!pip install -r OpenNMT-py/requirements.opt.txt

  Cloning git://github.com/NVIDIA/apex.git (to revision 700d6825e205732c1d6be511306ca4e595297070) to /tmp/pip-req-build-we_k43j8
  Running command git clone -q git://github.com/NVIDIA/apex.git /tmp/pip-req-build-we_k43j8
  Running command git checkout -q 700d6825e205732c1d6be511306ca4e595297070
  Running command git submodule update --init --recursive -q
     |████████████████████████████████| 3.3MB 14.1MB/s 
     |████████████████████████████████| 20.2MB 122kB/s 
     |████████████████████████████████| 61kB 10.0MB/s 
     |████████████████████████████████| 1.1MB 41.0MB/s 
  Created wheel for pyrouge: filename=pyrouge-0.1.3-cp36-none-any.whl size=191613 sha256=b2df2d67b37be6cb51badb7e11952a95db0d647960f388b7de6f1b13dc8ee22f
  Stored in directory: /root/.cache/pip/wheels/75/d3/0c/e5b04e15b6b87c42e980de3931d2686e14d36e045058983599
  Created wheel for apex: filename=apex-0.1-cp36-none-any.whl size=192741 sha256=975945dcaee68746814e88c5b70743409526aee554ad93b3dbd9948a7573bc3a
  Stored in d

In [ ]:
!pip install -U git+https://github.com/pltrdy/pyrouge

  Cloning https://github.com/pltrdy/pyrouge to /tmp/pip-req-build-o0fx18vg
  Running command git clone -q https://github.com/pltrdy/pyrouge /tmp/pip-req-build-o0fx18vg
  Created wheel for pyrouge: filename=pyrouge-0.1.3-cp36-none-any.whl size=191920 sha256=4d0b20f2e92e84aa4ea8b4e7b6ad117c4941ebebec69f7f5ce67496404c7d7e6
  Stored in directory: /tmp/pip-ephem-wheel-cache-txgcz6si/wheels/d7/6a/52/2534a6d70b54de6e94933267ff415b0b85cdb83836af9f74c3
Successfully built pyrouge
  Found existing installation: pyrouge 0.1.3
    Uninstalling pyrouge-0.1.3:
      Successfully uninstalled pyrouge-0.1.3


In [ ]:
!git clone https://github.com/pltrdy/files2rouge.git

fatal: destination path 'files2rouge' already exists and is not an empty directory.


In [ ]:
cd files2rouge

/content/drive/.shortcut-targets-by-id/1zJEzU7AONU-4f_78lpneSWk_NcBn0nda/DSGA_1011_NLP_Project/Data/CNN Daily Mail/files2rouge


In [ ]:
!pip install configargparse


     |████████████████████████████████| 51kB 7.8MB/s 
  Created wheel for configargparse: filename=ConfigArgParse-1.2.3-cp36-none-any.whl size=19329 sha256=10a06d94e80d54a6969e1a221bd74f3b9cdd5ef4ff68fd9fa884365119dbc765
  Stored in directory: /root/.cache/pip/wheels/bd/d6/53/034032da9498bda2385cd50a51a289e88090b5da2d592b1fdf
Successfully built configargparse


In [ ]:
!python OpenNMT-py/build_vocab.py \
 -data /content/drive/MyDrive/DSGA_1011_NLP_Project/Data/CNN Daily Mail/ \
 -train_src /content/drive/MyDrive/DSGA_1011_NLP_Project/Data/CNN Daily Mail/train.txt.src \
 -train_tgt /content/drive/MyDrive/DSGA_1011_NLP_Project/Data/CNN Daily Mail/train.txt.tgt.tagged \
 -valid_src /content/drive/MyDrive/DSGA_1011_NLP_Project/Data/CNN Daily Mail/valid.txt.src \
 -valid_tgt /content/drive/MyDrive/DSGA_1011_NLP_Project/Data/CNN Daily Mail/valid.txt.tgt.tagged \
 -src_vocab /content/drive/MyDrive/DSGA_1011_NLP_Project/Data/CNN Daily Mail/PREPROCESSED/example.vocab.src \
 -tgt_vocab /content/drive/MyDrive/DSGA_1011_NLP_Project/Data/CNN Daily Mail/PREPROCESSED/example.vocab.tgt \
 -save_data /content/drive/MyDrive/DSGA_1011_NLP_Project/Data/CNN Daily Mail/PREPROCESSED \
 -src_seq_length 400 \
 -tgt_seq_length 100 \
 -dynamic_dict \
 -share_vocab \
 -shard_size 200000

Traceback (most recent call last):
  File "OpenNMT-py/build_vocab.py", line 6, in <module>
    main()
  File "/content/drive/.shortcut-targets-by-id/1zJEzU7AONU-4f_78lpneSWk_NcBn0nda/DSGA_1011_NLP_Project/Data/CNN Daily Mail/OpenNMT-py/onmt/bin/build_vocab.py", line 66, in main
    build_vocab_main(opts)
  File "/content/drive/.shortcut-targets-by-id/1zJEzU7AONU-4f_78lpneSWk_NcBn0nda/DSGA_1011_NLP_Project/Data/CNN Daily Mail/OpenNMT-py/onmt/bin/build_vocab.py", line 23, in build_vocab_main
    ArgumentParser.validate_prepare_opts(opts, build_vocab_only=True)
  File "/content/drive/.shortcut-targets-by-id/1zJEzU7AONU-4f_78lpneSWk_NcBn0nda/DSGA_1011_NLP_Project/Data/CNN Daily Mail/OpenNMT-py/onmt/utils/parse.py", line 131, in validate_prepare_opts
    cls._validate_data(opt)
  File "/content/drive/.shortcut-targets-by-id/1zJEzU7AONU-4f_78lpneSWk_NcBn0nda/DSGA_1011_NLP_Project/Data/CNN Daily Mail/OpenNMT-py/onmt/utils/parse.py", line 29, in _validate_data
    for cname, corpus in corpora.

In [ ]:
onmt_build_vocab cnn_dm.yaml -n_sample -1

SyntaxError: ignored

In [ ]:
cd OpenNMT-py/

/content/drive/.shortcut-targets-by-id/1zJEzU7AONU-4f_78lpneSWk_NcBn0nda/DSGA_1011_NLP_Project/Data/cnndm/OpenNMT-py


In [ ]:
ls

available_models/  data/                   LICENSE.md            setup.py
build_vocab.py     docs/                   onmt/                 tools/
CHANGELOG.md       examples/               README.md             train.py
config/            floyd_requirements.txt  requirements.opt.txt  translate.py
CONTRIBUTING.md    floyd.yml               server.py


In [ ]:
pip install --upgrade git+https://github.com/OpenNMT/OpenNMT-py

  Cloning https://github.com/OpenNMT/OpenNMT-py to /tmp/pip-req-build-awth1vay
  Running command git clone -q https://github.com/OpenNMT/OpenNMT-py /tmp/pip-req-build-awth1vay
  Created wheel for OpenNMT-py: filename=OpenNMT_py-2.0.0rc2-cp36-none-any.whl size=199830 sha256=2b4ebe0de7c8ba56366ad003b354309661ac2edec0c0f28b93587551a3de9c11
  Stored in directory: /tmp/pip-ephem-wheel-cache-0p_vuemc/wheels/fd/c6/1b/c010315aed669f198a2e8b222221e3e379b72c456cc7a95498
Successfully built OpenNMT-py
  Found existing installation: OpenNMT-py 2.0.0rc2
    Uninstalling OpenNMT-py-2.0.0rc2:
      Successfully uninstalled OpenNMT-py-2.0.0rc2


In [ ]:
!!pip install git+https://github.com/pytorch/text

['Collecting git+https://github.com/pytorch/text',
 '  Cloning https://github.com/pytorch/text to /tmp/pip-req-build-msh0zs93',
 '  Running command git clone -q https://github.com/pytorch/text /tmp/pip-req-build-msh0zs93',
 '  Running command git submodule update --init --recursive -q',
 'Requirement already satisfied: tqdm in /usr/local/lib/python3.6/dist-packages (from torchtext==0.9.0a0+d6d7f20) (4.41.1)',
 'Requirement already satisfied: requests in /usr/local/lib/python3.6/dist-packages (from torchtext==0.9.0a0+d6d7f20) (2.23.0)',
 'Requirement already satisfied: torch in /usr/local/lib/python3.6/dist-packages (from torchtext==0.9.0a0+d6d7f20) (1.7.0+cu101)',
 'Requirement already satisfied: numpy in /usr/local/lib/python3.6/dist-packages (from torchtext==0.9.0a0+d6d7f20) (1.18.5)',
 'Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.6/dist-packages (from requests->torchtext==0.9.0a0+d6d7f20) (2.10)',
 'Requirement already satisfied: certifi>=2017.4.17 in /usr/

In [ ]:
!onmt_build_vocab -config cnndm.yaml -n_sample -1

^C


In [ ]:
ls

files2rouge/   run/                 train.txt.src         val.txt.tgt.tagged
OpenNMT-py/    test.txt.src         train.txt.tgt.tagged
PREPROCESSED/  test.txt.tgt.tagged  val.txt.src


In [ ]:
cd ..

/content/drive/.shortcut-targets-by-id/1zJEzU7AONU-4f_78lpneSWk_NcBn0nda/DSGA_1011_NLP_Project/Data


In [ ]:
!onmt_translate -gpu 0 \
               -batch_size 20 \
               -beam_size 10 \
               -model run/pretrained_models/lstm.pt \
               -src test.txt.src \
               -output pred_cnndm.out \
               -min_length 35 \
               -verbose \
               -stepwise_penalty \
               -coverage_penalty summary \
               -beta 5 \
               -length_penalty wu \
               -alpha 0.9 \
               -verbose \
               -block_ngram_repeat 3 \
               -ignore_when_blocking "." "</t>" "<t>"

Traceback (most recent call last):
  File "/usr/local/bin/onmt_translate", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.6/dist-packages/onmt/bin/translate.py", line 47, in main
    translate(opt)
  File "/usr/local/lib/python3.6/dist-packages/onmt/bin/translate.py", line 18, in translate
    translator = build_translator(opt, logger=logger, report_score=True)
  File "/usr/local/lib/python3.6/dist-packages/onmt/translate/translator.py", line 29, in build_translator
    fields, model, model_opt = load_test_model(opt)
  File "/usr/local/lib/python3.6/dist-packages/onmt/model_builder.py", line 92, in load_test_model
    opt.gpu)
  File "/usr/local/lib/python3.6/dist-packages/onmt/model_builder.py", line 131, in build_base_model
    src_field = fields["src"]
TypeError: list indices must be integers or slices, not str


In [ ]:
!pip install torch==1.6.0+cu101 torchvision==0.7.0+cu101 -f https://download.pytorch.org/whl/torch_stable.html

Looking in links: https://download.pytorch.org/whl/torch_stable.html
     |████████████████████████████████| 708.0MB 25kB/s 
     |████████████████████████████████| 5.9MB 4.2MB/s 
  Found existing installation: torch 1.6.0
    Uninstalling torch-1.6.0:
      Successfully uninstalled torch-1.6.0
  Found existing installation: torchvision 0.8.1+cu101
    Uninstalling torchvision-0.8.1+cu101:
      Successfully uninstalled torchvision-0.8.1+cu101


In [ ]:
!nvidia-smi


Fri Dec  4 04:48:12 2020       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 455.38       Driver Version: 418.67       CUDA Version: 10.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla T4            Off  | 00000000:00:04.0 Off |                    0 |
| N/A   48C    P8    10W /  70W |      0MiB / 15079MiB |      0%      Default |
|                               |                      |                 ERR! |
+-------------------------------+----------------------+----------------------+
                                                                               
+------

In [ ]:
!python OpenNMT-py/translate.py -gpu 1 \
 -batch_size 16 \
 -beam_size 5 \
 -model run/pretrained_models/transformers.pt \
 -src remaining_test.txt.src \
 -output test_remaining.pred \
 -min_length 6 \
 -verbose \
 -stepwise_penalty \
 -coverage_penalty summary \
 -beta 5 \
 -length_penalty wu \
 -alpha 0.9 \
 -block_ngram_repeat 3 \
 -ignore_when_blocking "." \
 -replace_unk

Output hidden; open in https://colab.research.google.com to view.

In [ ]:
!python OpenNMT-py/translate.py -model Data/cnndm/run/pretrained_models/lstm.pt -src Data/cnndm/test.txt.src -output Data/cnndm/pred.txt -replace_unk -verbose

Traceback (most recent call last):
  File "OpenNMT-py/translate.py", line 6, in <module>
    main()
  File "/content/drive/.shortcut-targets-by-id/1zJEzU7AONU-4f_78lpneSWk_NcBn0nda/DSGA_1011_NLP_Project/OpenNMT-py/onmt/bin/translate.py", line 47, in main
    translate(opt)
  File "/content/drive/.shortcut-targets-by-id/1zJEzU7AONU-4f_78lpneSWk_NcBn0nda/DSGA_1011_NLP_Project/OpenNMT-py/onmt/bin/translate.py", line 18, in translate
    translator = build_translator(opt, logger=logger, report_score=True)
  File "/content/drive/.shortcut-targets-by-id/1zJEzU7AONU-4f_78lpneSWk_NcBn0nda/DSGA_1011_NLP_Project/OpenNMT-py/onmt/translate/translator.py", line 29, in build_translator
    fields, model, model_opt = load_test_model(opt)
  File "/content/drive/.shortcut-targets-by-id/1zJEzU7AONU-4f_78lpneSWk_NcBn0nda/DSGA_1011_NLP_Project/OpenNMT-py/onmt/model_builder.py", line 92, in load_test_model
    opt.gpu)
  File "/content/drive/.shortcut-targets-by-id/1zJEzU7AONU-4f_78lpneSWk_NcBn0nda/DSGA_10

In [ ]:
!onmt_translate -model run/pretrained_models/lstm.pt -src test.txt.src -output run/pred_1000.txt -gpu 0 -verbose

Traceback (most recent call last):
  File "/usr/local/bin/onmt_translate", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.6/dist-packages/onmt/bin/translate.py", line 47, in main
    translate(opt)
  File "/usr/local/lib/python3.6/dist-packages/onmt/bin/translate.py", line 18, in translate
    translator = build_translator(opt, logger=logger, report_score=True)
  File "/usr/local/lib/python3.6/dist-packages/onmt/translate/translator.py", line 29, in build_translator
    fields, model, model_opt = load_test_model(opt)
  File "/usr/local/lib/python3.6/dist-packages/onmt/model_builder.py", line 92, in load_test_model
    opt.gpu)
  File "/usr/local/lib/python3.6/dist-packages/onmt/model_builder.py", line 131, in build_base_model
    src_field = fields["src"]
TypeError: list indices must be integers or slices, not str


In [ ]:
!onmt_train -config rnncnn.yaml

[2020-11-29 21:11:01,052 INFO] Missing transforms field for cnndm data, set to default: [].
[2020-11-29 21:11:01,053 WARNING] Corpus cnndm's weight should be given. We default it to 1 for you.
[2020-11-29 21:11:01,053 INFO] Missing transforms field for valid data, set to default: [].
[2020-11-29 21:11:01,053 INFO] Parsed 2 corpora from -data.
[2020-11-29 21:11:01,214 INFO] Get special vocabs from Transforms: {'src': set(), 'tgt': set()}.
[2020-11-29 21:11:01,215 INFO] Loading vocab from text file...
[2020-11-29 21:11:01,215 INFO] Loading src vocabulary from cnndm/run/example/vocab.src
[2020-11-29 21:11:03,346 INFO] Loaded src vocab has 722715 tokens.
[2020-11-29 21:11:03,775 INFO] Loading tgt vocabulary from cnndm/run/example/tgt_vocab.tgt
[2020-11-29 21:11:04,528 INFO] Loaded tgt vocab has 197771 tokens.
[2020-11-29 21:11:04,630 INFO] Building fields with vocab in counters...
[2020-11-29 21:11:05,031 INFO]  * tgt vocab size: 50004.
[2020-11-29 21:11:06,351 INFO]  * src vocab size: 500

In [ ]:
!git checkout a0095fa2

error: Your local changes to the following files would be overwritten by checkout:
	examples/scripts/prepare_wmt_data.sh
	onmt/bin/average_models.py
	onmt/bin/release_model.py
	onmt/bin/server.py
	onmt/bin/translate.py
	onmt/tests/pull_request_chk.sh
	onmt/tests/rebuild_test_models.sh
	tools/average_models.py
	tools/embeddings_to_torch.py
Please commit your changes or stash them before you switch branches.
Aborting


In [ ]:
!git clone https://github.com/OpenNMT/OpenNMT-py.git

fatal: destination path 'OpenNMT-py' already exists and is not an empty directory.


In [ ]:
cd OpenNMT-py

/content/drive/.shortcut-targets-by-id/1zJEzU7AONU-4f_78lpneSWk_NcBn0nda/DSGA_1011_NLP_Project/Data/cnndm/OpenNMT-py


In [ ]:
!nvidia-smi

Tue Nov 24 20:57:31 2020       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 455.38       Driver Version: 418.67       CUDA Version: 10.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla T4            Off  | 00000000:00:04.0 Off |                    0 |
| N/A   32C    P8     9W /  70W |      0MiB / 15079MiB |      0%      Default |
|                               |                      |                 ERR! |
+-------------------------------+----------------------+----------------------+
                                                                               
+-------

In [ ]:
ls

 ACL_2020_Proceedings_Template.pdf   Gig_colab.ipynb
 CNN_Daily_Mail.ipynb                OpenNMT-py/
 Data/                               OpenNMT_Translation_TedTalks.ipynb
'DSGA1011 NLP Project.gdoc'          Ted_Talks_EDA.ipynb
 EUROPARL_INSPECTION.ipynb           Ted_Talks_EDA_v2.ipynb


In [ ]:
!python OpenNMT-py/preprocess.py \
 -train_src Data/cnndm/train.txt.src \
 -train_tgt Data/cnndm/train.txt.tgt \
 -valid_src Data/cnndm/valid.txt.src \
 -valid_tgt Data/cnndm/valid.txt. tgt\
 -save_data Data/cnndm/PREPROCESSED \
 -src_seq_length 10000 \
 -dynamic_dict \
 -share_vocab \
 -shard_size 200000

python3: can't open file 'OpenNMT-py/preprocess.py': [Errno 2] No such file or directory


In [ ]:
cd OpenNMT-py

/content/drive/.shortcut-targets-by-id/1zJEzU7AONU-4f_78lpneSWk_NcBn0nda/DSGA_1011_NLP_Project/OpenNMT-py


In [ ]:
ls

available_models/  data/                   LICENSE.md            setup.py
build/             dist/                   onmt/                 tools/
build_vocab.py     docs/                   OpenNMT_py.egg-info/  train.py
CHANGELOG.md       examples/               README.md             translate.py
config/            floyd_requirements.txt  requirements.opt.txt
CONTRIBUTING.md    floyd.yml               server.py


In [ ]:
!pip install configargparse
#!pip install torch==1.7.0
!pip install git+https://github.com/pytorch/text

  Cloning https://github.com/pytorch/text to /tmp/pip-req-build-2tpjch5t
  Running command git clone -q https://github.com/pytorch/text /tmp/pip-req-build-2tpjch5t
  Running command git submodule update --init --recursive -q
  Created wheel for torchtext: filename=torchtext-0.9.0a0+d6d7f20-cp36-cp36m-linux_x86_64.whl size=7049330 sha256=7544d909ac7d5b07e496c151856d5557ec016b2ce1b763015a2140047b5de769
  Stored in directory: /tmp/pip-ephem-wheel-cache-fyz93r70/wheels/73/14/71/ed033fd999ae4933e17df3e91be2014e61c2f312a88a164ff5
Successfully built torchtext
  Found existing installation: torchtext 0.3.1
    Uninstalling torchtext-0.3.1:
      Successfully uninstalled torchtext-0.3.1


In [13]:
!git clone --recursive https://github.com/pltrdy/files2rouge.git

shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory
fatal: could not create work tree dir 'files2rouge': No such file or directory


In [7]:
cd files2rouge

/content/drive/.shortcut-targets-by-id/1zJEzU7AONU-4f_78lpneSWk_NcBn0nda/DSGA_1011_NLP_Project/Data/cnndm/files2rouge


In [12]:
!git checkout v0.2

shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory
fatal: Unable to read current working directory: No such file or directory


In [9]:
ls

Dockerfile    files2rouge/  MANIFEST.in  setup.py
experiments/  LICENSE       README.md    setup_rouge.py


In [11]:
!files2rouge ../cnndm/run/test.pred ../cnndm/test.txt.tgt.tagged 

/bin/bash: files2rouge: command not found
